vulcanai2/models/metrics.py

```diff
@@ -28,81 +28,66 @@ class Metrics(object):
-
-    #     self.mat += confusion_matrix(temp_targets, temp_predictions, labels=self.list_classes)
-
-    def get_score(self, predictions, targets, metric='accuracy'):
-        """
-        Calculate some defined score given predictions and targets.
 
+    def get_score(self, targets, predictions, metrics='accuracy', average=None,
+                  class_converted=False):
+        """
+        Calculate the provided metrics given some targets and predictions
         Parameters
         ----------
-        predictions : torch.Tensor
-            Network output of shape [batch, num_classes].
-        targets : torch.LongTensor
-            The truth values of shape [batch].
-        metric : str
-            The metric to calculate and return.
-
-        Returns
+        targets: numpy.ndarray of integers
+            the target values
+        predictions: numpy.ndarray of integers
+            the predicted values
+        metrics: list of strings
+            the strings definition the "get_"... functions to call
+        average: string,
+            [None, ‘binary’ (default), ‘micro’, ‘macro’, ‘samples’, ‘weighted’]
+            This parameter is required for multiclass/multilabel targets.
+            If None, the scores for each class are returned.
+            Otherwise, this determines the type of averaging performed on the
+            data.See scikit learn
+        class_converted: binary. default False
+            True if raw_predictions have already been converted using
+            extract_class_labels
+            False if raw_predictions are used
+
+        Returns dict
+            Metric name : values returned by the metric functions
         -------
-        score : float
-            The specified metric to calculate.
 
         """
-        if metric == 'accuracy':
-            # TODO: Use get_class
-            max_index = predictions.max(dim=1)[1]
-            correct = (max_index == targets).sum() # TODO: this doesn't seem correct
-            accuracy = int(correct.data) / len(targets)
-            return accuracy
-        else:
-            raise NotImplementedError(
-                'Metric {} not available.'.format(metric))
 
-    # TODO: class # should correspond with self.num_class
-    # noinspection PyMethodMayBeStatic
-    def get_class(self, in_matrix):
+        double_parametered_functions = ["get_accuracy"]
+
+        if not class_converted:
+            predictions = self.extract_class_labels(predictions)
+
+        if isinstance(metrics, str):
+            metrics = [metrics]
+
+        results_dict = {}
+
+        for metric in metrics:
+            method_name = "get_" + metric
+            method = getattr(Metrics, method_name)
+            try:
+                if method_name in double_parametered_functions:
+                    res = method(targets, predictions)
+                else:
+                    res = method(targets, predictions, average)
+
+            except AttributeError:
```

Comment:
Should we also try to catch for some sort of not callable error?

Reply:
add in TypeError and description